GOVERNANCE.md

```diff
@@ -0,0 +1,150 @@
+# Abstract
+
+The purpose of this document is to formalize the governance process used by the
+`zarr` project, to clarify how decisions are made and how the various
+elements of our community interact.
+
+This is a consensus-based community project. Anyone with an interest in the
+project can join the community, contribute to the project design, and
+participate in the decision making process. This document describes how that
+participation takes place, how to find consensus, and how deadlocks are
+resolved.
+
+# Roles And Responsibilities
+
+## The Community
+
+The zarr community consists of anyone using or working with the project
+in any way.
+
+## Contributors
+
+A community member can become a contributor by interacting directly with the
+project in concrete ways, such as:
+
+- proposing, discussing, or reviewing a change to the code, documentation, or specification
+  via a GitHub pull request to any https://github.com/zarr-developers repository;
+- reporting a GitHub issue
+  to any https://github.com/zarr-developers repository;
+- discussing examples or usage issues under the #zarr tag on StackOverflow 
+  (like "How do I do X in zarr?").
+
+among other possibilities. Any community member can become a contributor, and
+all are encouraged to do so. By contributing to the project, community members
+can directly help to shape its future.
+
+Contributors are encouraged to read the [contributing guide](http://zarr.readthedocs.io/en/stable/contributing.html)
+for information about contributing to Zarr.
+
+## Core developers
+
+Core developers are community members that have demonstrated continued
+commitment to the project through ongoing contributions. They
+have shown they can be trusted to maintain zarr with care. Becoming a
+core developer allows contributors to merge approved pull requests, cast votes
+for and against merging a pull-request, and be involved in deciding major
+changes to the API, and thereby more easily carry on with their project related
+activities. Core developers appear as organization members on the zarr
+[GitHub organization](https://github.com/orgs/zarr-developers/people) and are on our
+[@zarr-developers/core-devs](https://github.com/orgs/zarr-developers/teams/core-devs) GitHub team. Core
+developers are expected to review code contributions while adhering to the
+[core developer guide](CORE_DEV_GUIDE.md). New core developers can be nominated
+by any existing core developer, and for details on that process see our core
+developer guide.
```

Comment:
If we included the details of how one can become a contributor and steering council member, then this document would be more self contained. I would be in favor of adding these details.

Reply:
I looked at a few core-dev documents and the modifications were going to be pretty substantial. I like the idea of working all of the content into the same file and perhaps starting with less and building up over time. Happy to get a start on that if there are no objections.

Reply:
Strike that. Was easier to get `CORE_DEV_GUIDE.md` as it stood into a form that was reviewable. Opinions still welcome on merging it into `GOVERNANCE.md`.